dd-sdk-android-tv/src/test/kotlin/com/datadog/android/tv/LeanbackViewAttributesProviderTest.kt

```diff
@@ -0,0 +1,377 @@
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeAttributes = forge.exhaustiveAttributes().toMutableMap()
+        fakeAction = Action(fakeActionId, fakeActionLabel1, fakeActionLabel2)
+        whenever(mockViewHolder.item).thenReturn(fakeAction)
+        testedAttributesProvider =
+            LeanbackViewAttributesProvider()
+    }
+
+    override fun createInstance(forge: Forge): LeanbackViewAttributesProvider {
+        return LeanbackViewAttributesProvider()
+    }
+
+    override fun createEqualInstance(
+        source: LeanbackViewAttributesProvider,
+        forge: Forge
+    ): LeanbackViewAttributesProvider {
+        return LeanbackViewAttributesProvider()
+    }
+
+    override fun createUnequalInstance(
+        source: LeanbackViewAttributesProvider,
+        forge: Forge
+    ): LeanbackViewAttributesProvider? = null
+
+    @Test
+    fun `M add action details W extractAttributes`() {
+        // Given
+        whenever(mockTarget.layoutParams).thenReturn(mock<RecyclerView.LayoutParams>())
+        whenever(mockRecyclerView.findContainingViewHolder(mockTarget)).thenReturn(
+            mockViewHolder
+        )
+        whenever(mockTarget.parent).thenReturn(mockRecyclerView)
+        val expectedExtraEntries = fakeAttributes.removeActionAttributes()
+
+        // When
+        testedAttributesProvider.extractAttributes(mockTarget, fakeAttributes)
+
+        // Then
+        assertThat(fakeAttributes).containsAllEntriesOf(expectedExtraEntries)
+        assertThat(fakeAttributes).containsAllEntriesOf(
+            mapOf(
+                LeanbackViewAttributesProvider.ACTION_TARGET_ACTION_ID to fakeAction.id,
+                LeanbackViewAttributesProvider.ACTION_TARGET_LABEL1 to fakeAction.label1,
+                LeanbackViewAttributesProvider.ACTION_TARGET_LABEL2 to fakeAction.label2,
+            )
+        )
+    }
+
+    @Test
+    fun `M not add action label1 W extractAttributes {Action Label 1 is null}`() {
+        // Given
+        fakeAction.label1 = null
+        whenever(mockTarget.layoutParams).thenReturn(mock<RecyclerView.LayoutParams>())
+        whenever(mockRecyclerView.findContainingViewHolder(mockTarget)).thenReturn(
+            mockViewHolder
+        )
+        whenever(mockTarget.parent).thenReturn(mockRecyclerView)
+        val expectedExtraEntries = fakeAttributes.removeActionAttributes()
+
+        // When
+        testedAttributesProvider.extractAttributes(mockTarget, fakeAttributes)
+
+        // Then
+        assertThat(fakeAttributes).containsAllEntriesOf(expectedExtraEntries)
+        assertThat(fakeAttributes).containsAllEntriesOf(
+            mapOf(
+                LeanbackViewAttributesProvider.ACTION_TARGET_ACTION_ID to fakeAction.id,
+                LeanbackViewAttributesProvider.ACTION_TARGET_LABEL2 to fakeAction.label2
+            )
+        )
+        assertThat(fakeAttributes)
+            .doesNotContainEntry(LeanbackViewAttributesProvider.ACTION_TARGET_LABEL1, null)
```

Comment:
we can probably simply use `doesNotContainKey` here?

Reply:
nope because that `key` can be added (even though the probability is very low) when the attributes are forged. It is better to actually check it together with the value.

Reply:
You can also constrain the way attributes are forged to avoid adding this key by mistake. Because if we change the implementation and somehow we add by mistake this attribute with a different value, your test won't be valid anymore